### PR TITLE
Add numeric compare for validation

### PR DESCRIPTION
**Pull request by Mabuchin**
Hi,
I needed numeric compare when pre-post validate such as cpu_use,memory_available.
for example, less than memory 50% is executed by commit_config.

* modified: _compare_getter_dict
* add method: compare_numeric
* add test: get_environment

### CODE REVIEW

File: test/unit/validate/mocked_data/non_strict_fail/validate.yml [hunk 1]
```diff
@@ -37,3 +37,10 @@
        - next_hop: 10.155.180.22
          outgoing_interface: "irb.0"
          protocol: "OSPF"
+
+- get_environment:
```
Comment (dbarrosop) [inline test/unit/validate/mocked_data/non_strict_fail/validate.yml]: Would you mind adding this as an example here:

https://napalm.readthedocs.io/en/latest/validate/index.html#documentation

**Comment by dbarrosop** [comment]
Sorry for the late response on this. Looks good so I will merge after we add a note to the documentation.

Thanks!

**Comment by Mabuchin** [comment]
Thanks!  I'll add a note to the documentation and send PR right away :-)

**Comment by Mabuchin** [comment]
Sent PR to napalm [#377](https://github.com/napalm-automation/napalm/pull/377)

**Comment by dbarrosop** [comment]
Great job, thanks!

**Pull request merged**